patches/tinyemu/tinyemu/slirp/tcp_subr.c

```diff
@@ -355,11 +357,16 @@ int tcp_fconnect(struct socket *so)
     /* We don't care what port we get */
     ret = connect(s,(struct sockaddr *)&addr,sizeof (addr));
 
+    if (ret == 0) {
+        soisfconnected(so);
+    } else
+    {
     /*
      * If it's not in progress, it failed, so we just return 0,
      * without clearing SS_NOFDREF
      */
-    soisfconnecting(so);
+      soisfconnecting(so);
+    }
```

Comment:
How is this change related to OOB? If this doesn't work, it rather feels like the socket doesn't support non-blocking `connect` (e.g. `select` on the socket doesn't set the writefd on ready).

Reply:
It's a combination of both as far as I can tell, OOB flags stopping WAMR from continuing connecting as well as non-blocking not working at all.